src/freertos_drivers/esp32/Esp32WiFiManager.cxx

```diff
@@ -0,0 +1,985 @@
+                match_found = true;
+            }
+            ipaddr = ipaddr->next;
+        }
+        res = res->next;
+    }
+
+    // free up the query results linked list.
+    mdns_query_results_free(results);
+
+    if(!match_found)
+    {
+        LOG(WARNING, "[mDNS] No matches found for service: %s.", service);
+        return EAI_AGAIN;
+    }
+
+    // return the resolved data to the caller
+    *addr = ai.release();
+    (*addr)->ai_addr = sa.release();
+
+    // successfully resolved an address, inform the caller
+    return 0;
+}
+
+// The functions below are not available via the standard ESP-IDF provided
+// API.
+
+/// Retrieves the IPv4 address from the ESP32 station interface.
+///
+/// @param ifap will hold the IPv4 address for the ESP32 station interface when
+/// successfully retrieved.
+/// @return zero for success, -1 for failure.
+int getifaddrs(struct ifaddrs **ifap)
+{
+    tcpip_adapter_ip_info_t ip_info;
+
+    /* start with something "safe" in case we bail out early */
+    *ifap = nullptr;
+
+    if(!tcpip_adapter_is_netif_up(TCPIP_ADAPTER_IF_STA))
+    {
+        // Station TCP/IP interface is not up
+        errno = ENODEV;
+        return -1;
+    }
+
+    // allocate memory for various pieces of ifaddrs
+    std::unique_ptr<struct ifaddrs> ia(new struct ifaddrs);
+    if (ia.get() == nullptr)
+    {
+        errno = ENOMEM;
+        return -1;
+    }
+    memset(ia.get(), 0, sizeof(struct ifaddrs));
+    std::unique_ptr<char[]> ifa_name(new char[6]);
+    if(ifa_name.get() == nullptr)
+    {
+        errno = ENOMEM;
+        return -1;
+    }
+    strcpy(ifa_name.get(), "wlan0");
+    std::unique_ptr<struct sockaddr> ifa_addr(new struct sockaddr);
+    if (ifa_addr == nullptr)
+    {
+        errno = ENOMEM;
+        return -1;
+    }
+    memset(ifa_addr.get(), 0, sizeof(struct sockaddr));
+
+    // retrieve TCP/IP address from the interface
+    tcpip_adapter_get_ip_info(TCPIP_ADAPTER_IF_STA, &ip_info);
+
+    // copy address into ifaddrs structure
+    struct sockaddr_in *addr_in = (struct sockaddr_in*)ifa_addr.get();
+    addr_in->sin_family = AF_INET;
+    addr_in->sin_addr.s_addr = ip_info.ip.addr;
+    ia.get()->ifa_next = nullptr;
+    ia.get()->ifa_name = ifa_name.release();
+    ia.get()->ifa_flags = 0;
+    ia.get()->ifa_addr = ifa_addr.release();
+    ia.get()->ifa_netmask = nullptr;
+    ia.get()->ifa_ifu.ifu_broadaddr = nullptr;
+    ia.get()->ifa_data = nullptr;
+
+    // report results
+    *ifap = ia.release();
+    return 0;
+}
+
+/// Frees memory allocated as part of the call to @ref getifaddrs.
+///
+/// @param ifa is the ifaddrs struct to be freed.
+void freeifaddrs(struct ifaddrs *ifa)
+{
+    while (ifa)
+    {
+        struct ifaddrs *next = ifa->ifa_next;
+
+        HASSERT(ifa->ifa_data == nullptr);
+        HASSERT(ifa->ifa_ifu.ifu_broadaddr == nullptr);
+        HASSERT(ifa->ifa_netmask == nullptr);
+
+        if (ifa->ifa_addr)
```

Comment:
delete nullptr is fine.

Reply:
copied from CC32xx, will drop condition checks since they will always be defined.